core/src/main/java/software/amazon/awssdk/RequestExecutionContext.java

```diff
@@ -78,17 +74,27 @@ public RequestConfig requestConfig() {
     }
 
     /**
-     * @return Request handlers to hook into request lifecycle.
+     * @return Execution interceptors to hook into execution lifecycle.
      */
-    public List<RequestHandler> requestHandlers() {
-        return Collections.unmodifiableList(requestHandlers);
+    public ExecutionInterceptorChain interceptorChain() {
```

Comment:
Is there any value to storing it as a chain instead of a composite ExecutionInterceptor? Seems more consistent with what we do for credentials and just more flexible overall.